src/commands/txCommands/pullAll.js

```diff
@@ -0,0 +1,73 @@
+const chalk = require('chalk');
+const child_process = require('child_process');
+const Rx = require('rxjs');
+const txlib = require('./util');
+const txPullResource = require('./util/pullResourceTrns');
+
+const getProjectResourcesList$ =
+	txlib.resources$
+		.flatMap(Rx.Observable.from)
+		.reduce((resources, resource) => [ ...resources, resource ], []);
+
+/**
+ * recursively goes through list of resources and pulls trn content down
+ */
+const pullResource = (resources, iteration, commitOnComplete) => {
+	console.log(chalk.cyan(`Starting tx:pull for '${resources[iteration]}'`));
+	txPullResource.pullResourceContent$(resources[iteration]).subscribe(null, null, () => {
+		console.log(chalk.green(`Completed tx:pull for '${resources[iteration]}'`));
+
+		const doNext = iteration+1 < resources.length;
+
+		// only do a git commit if this is requested by the user when starting tx pull
+		if (commitOnComplete) {
+			// do an initial check to see if there is actually something to commit
+			child_process.exec('git status', (error, stdout, stderr) => {
+				// if 'nothing to commit' is NOT in `stdout` make a new commit
+				// with resource name in commit message
+				if (stdout.includes('nothing to commit') === false) {
```

Comment:
not sure how i feel about this check... couldn't think of a better way to avoid unneeded commits it the logs... but might also be overthinking in that it should be avoided

Reply:
Seems ok. In general it's nice to put the "true" case first just because it's a little easier to scan, so you might want to either re-arrange this blocks or do something like `if (stdout.includes('modified:')) {`